lmms_eval/protocol.py

```
@@ -114,3 +127,122 @@ def encode_image(self, image: Union[Image.Image, str]):
+        if nframes is not None:
+            video_dict["nframes"] = nframes
+            video_dict.pop("fps")
+
+        return fetch_video(video_dict)
+
+    def process_video_fast(
+        self,
+        video_path: str,
+        max_pixels: int,
+        fps: int,
+        nframes: int = None,
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)